lib/mixins/lazy-upgrade-mixin.html

```diff
@@ -0,0 +1,131 @@
+<!--
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<link rel="import" href="../utils/boot.html">
+<link rel="import" href="../utils/mixin.html">
+<link rel="import" href="../utils/render-status.html">
+<script>
+  (function(){
+    'use strict';
+
+  /**
+   * Element class mixin that provides API for lazily upgrading Polymer Elements.
+   *
+   * This mixin can be used with the `disable-upgrade` attribute to lazily upgrade elements
+   * that are descendants of this element. This is useful for elements that need to display
+   * many elements.
+   *
+   * When upgrading elements, the mixin will upgrade as many as it can within the budget
+   * set by `_lazyUpgradeBudget`, which defaults to 16ms.
+   *
+   * When finished processing lazy-upgrade elements, this mixin will set `_lazyUpgrading`
+   * to false on the instance.
+   *
+   * @polymerMixin
+   * @memberof Polymer
+   * @summary Element class mixin that provides API for lazily upgrading Polymer elements
+   */
+    Polymer.LazyUpgradeMixin = Polymer.dedupingMixin(function(base) {
+
+      const LAZY_UPGRADE = 'lazy-upgrade';
+      const DISABLE_UPGRADE = 'disable-upgrade';
+      const LAZY_UPGRADE_QUERY = `[${LAZY_UPGRADE}]`;
+      const LAZY_UPGRADE_BUDGET = 16;
+
+      function sortCandidates(a, b) {
+        let orderA = parseInt(a.getAttribute(LAZY_UPGRADE), 10) || 0;
+        let orderB = parseInt(b.getAttribute(LAZY_UPGRADE), 10) || 0;
+        return orderA - orderB;
+      }
+
+      function findCandidates(root) {
+        let candidates = Array.from(root.querySelectorAll(LAZY_UPGRADE_QUERY));
+        candidates.sort(sortCandidates);
+        return candidates;
+      }
+
+      function upgradeNode(node) {
+        if (node.localName === 'dom-if') {
+          node.if = true;
+        }
+        node.removeAttribute(DISABLE_UPGRADE);
+        node.removeAttribute(LAZY_UPGRADE);
+      }
+
+      function eventInCurrentScope(scope, event) {
+        return event.target === event.composedPath()[0];
+      }
+
+      /** @polymerMixinClass */
+      return class LazyUpgrade extends base {
+        // add a configurable
+        static get properties() {
+          return {
+            /**
+             * Instance-level property for configuring the frame budget for lazy-upgrading elements.
+             * Defaults to 16ms
+             */
+            _lazyUpgradeBudget: {
```

Comment:
I'm not sure I understand the `targetFramerate` API in dom-repeat. Is it the desired # of elements to be stamped every second?